Create and update user endpoints

04HuynhM opened the pull request:
Title.

Commit:
Endpoints added for create and update user

 CreateUser takes:
	firstName,
	lastName,
        email,
        password,
        phone,
        address,
        city,
        country,
        gender,
        date_of_birth

UpdateUser takes:
        uid,
        updatedType (one of the above),
        value

User fields are subject to change.

Commit:
Added validation and suitable responses

Validation for incoming form data added
Suitable responses for success and failure implemented

Commit:
Merge branch 'create-update-user' of github.com:alassadi/SmartHomeServer into create-update-user

Commit:
Merge branch 'development' of github.com:alassadi/SmartHomeServer into create-update-user

thefnox commented on functions/index.js:
```diff
@@ -5,13 +5,15 @@ admin.initializeApp(functions.config().firebase);
   ...rooms,
-  ...devices
+  ...devices,
+  ...users
```
You're about to haver a merge conflict with the other pr

thefnox commented on functions/users/index.js:
```diff
@@ -0,0 +1,102 @@
+        const data = req.body;
+        if (!checkUserData(data)) {
+            return res.status(400).json({
+                message: 'Missing parameters ' +
```
There's no need for concatenation here, this is a single constant

thefnox commented on functions/users/index.js:
```diff
@@ -0,0 +1,102 @@
+            disabled: false
+        })
+            .then((userRecord) => {
+                console.log("User created with UID of: " + userRecord.uid);
```
Prefer template strings to concatenation, 
```suggestion
                console.log(`User created with UID of: ${userRecord.uid}`);
```

thefnox commented on functions/users/index.js:
```diff
@@ -0,0 +1,102 @@
+    } else {
+        const data = req.body;
+        dbref.child('Users').child(data.uid).update({
+            [data.updatedType] : data.value.toString()
```
Yikes, this allows the user to modify all attributes? Shouldn't we restrict to specific ones?

04HuynhM replied:
Yeah, good point. We only need to restrict email though.

thefnox commented on functions/users/index.js:
```diff
@@ -0,0 +1,102 @@
+                })
+            })
+            .catch((error) => {
+                console.log("Error creating user: ", error);
```
Same here

Commit:
Placed references inside functions

Moved database references inside functions just in case
they are mutable.

Also removed some unnecessary else statements

thefnox committed:
Merge branch 'development' into create-update-user
